credentials/apps/edx_credentials_extensions/edly_credentials_app/api/fields.py

```diff
@@ -0,0 +1,24 @@
+import base64
+
+from django.core.files.base import ContentFile
+from rest_framework import serializers
+
+
+class Base64ImageField(serializers.ImageField):
+    """
+    Custom serializer field to ingest base64 encoded image
+    """
+
+    def to_internal_value(self, data):
+        """
+        Save base 64 encoded images
+        """
+        if not data:
+            return None
+
+        if isinstance(data, str) and data.startswith('data:image'):
+            file_format, imgstr = data.split(';base64,')  # format ~= data:image/X;base64,/xxxyyyzzz/
+            ext = file_format.split('/')[-1]
+            data = ContentFile(base64.b64decode(imgstr), name='tmp.' + ext)
```

Comment:
Have we checked if the file could be overridden during upload and store process on credentials services as we are storing files with same name?

Reply:
Looks like `ContentFile` handles that automatically.
<img width="256" alt="Screenshot 2020-11-30 at 1 32 09 PM" src="https://user-images.githubusercontent.com/15142776/100586114-7461ec80-3310-11eb-9619-d3537c0ad331.png">